File: 프로그래머스/lv3/49189. 가장 먼 노드/가장 먼 노드.py
from collections import deque

def solution(n, edge):
    answer = 1
    
    visit = [0] * (n+1)
    graph = [[] for _ in range(n+1)]
    
    for s, e in edge:
        graph[s].append(e)
        graph[e].append(s)
    
    visit[1] = 1
    q = deque([1])
    
    while q:
        now = q.popleft()
        for next in graph[now]:
            if visit[next] == 0:
                visit[next] = visit[now] + 1
                q.append(next)
    maxv = max(visit)
    answer = max(answer, visit.count(maxv))
    return answer

n = 6
edge = [[3, 6], [4, 3], [3, 2], [1, 3], [1, 2], [2, 4], [5, 2]]
print(solution(n, edge))

File: 프로그래머스/3/49189. 가장 먼 노드/가장 먼 노드.js
function solution(n, edge) {
    var answer = 0;
    let visit = Array.from(Array(n+1), ()=>(0));
    let graph = Array.from(Array(n+1), ()=>(Array()));
    
    for ([s, e] of edge) {
        graph[s].push(e);
        graph[e].push(s);
    }
    q = [1];
    visit[1] = 1;
    while (q.length > 0) {
        let now = q.shift();
        if (graph[now]){
            graph[now].forEach((el) => {
                if (visit[el] === 0) {
                    visit[el] = visit[now] + 1;
                    q.push(el);
                };
            });
        }
    }

    let maxValue = Math.max(...visit)
    answer = visit.reduce((cnt, el) => cnt + (maxValue===el), 0);
    
    return answer;
}